server/router/rider.ts

```diff
@@ -60,6 +62,24 @@ router.get('/:id/favorites', validateUser('User'), (req, res) => {
   });
 });
 
+// Get current/soonest ride (within next 30 min) of rider, if exists
+router.get('/:id/currentride', validateUser('Rider'), (req, res) => {
+  const { params: { id } } = req;
+  db.getById(res, Rider, id, tableName, () => {
+    const now = moment().toISOString();
+    const end = moment().add(30, 'minutes').toISOString();
```

Comment:
It might be better style to make a separate variable for the start window instead of hardcoding "30" in case it is changed later, but since it is commented that it's within the next 30 min it is probably fine